src/conv/k-g/k_parser.cpp

```diff
@@ -348,6 +352,22 @@ bool parse_k
 			++sectionLinesRead;
 			break;
 		    }
+		    case KState::PART_ADAPTIVE_FAILURE: {
+			if (tokens.size() == 0) {

```

Comment:
There are a bit to many tests for *tokens.size()* here. One `if (tokens.size() < 2) {` then error is enough. If there are more than 2, the surplus will be ignores, as in the other states.